index.js

```diff
@@ -198,6 +198,9 @@ function requestAsEventEmitter(opts) {
 						const uploadEventFrequency = 150;
 
 						progressInterval = setInterval(() => {
+							if (socket.destroyed) {
+								clearInterval(progressInterval);
```

Comment:
Shouldn't it return here? I don't think you want to run the below code if the socket is destroyed.